src/pybind/mgr/ansible/module.py

```diff
@@ -455,6 +569,45 @@ def process_playbook_result(inventory_events, ar_client, playbook_uuid):
 
     return result
 
+def process_hosts_ls(host_ls_json):
+    """ Format the output of host ls call
+
+    Ansible runner service provides the following json:
+            {
+            "status": "OK",
+            "msg": "",
+            "data": {
+                "hosts": [
+                    "192.168.121.20",
+                    "192.168.121.170",
+                    "192.168.121.213",
+                    "192.168.121.212"]
+                }
+            }
+    The output provide is like:
+
+    Hosts
+    192.168.121.20
+    192.168.121.170
+    192.168.121.213
+    192.168.121.212
+    """
+    output = ""
+
+    try:
+        json_resp = json.loads(host_ls_json)
+
+        row_format = "{0:<25}\n"
+        output += row_format.format("Hosts")
+        for host in json_resp["data"]["hosts"]:
+            output += row_format.format(host)
+
+    except Exception:
```

Comment:
Catching `Exception` is very broad here. Which exceptions do you wan to catch?

Reply:
Added expected exceptions to catch